### PR TITLE
Add support for video files

### LINKED ISSUE #17: Add support for animation_url
For video NFTs we are going to support the [opensea standard](https://docs.opensea.io/docs/metadata-standards) for the metadata and let the workflow define both animation_url and image.

hence the generateMetadata in metadata module should accept multimedia files as well as image files and upload both of them to the ipfs and create metadata as:

{
...
image:<url to image resource>
animation_url:<url to multi-media resource>
}

The workflow should also allow specifying a multimedia path as well as an image path. The paths can be said as described in #16

### PR DESCRIPTION
**Pull request by jsidorenko**
Closes #17 
Based on #18

### CODE REVIEW

File: src/workflow/metadata.js [hunk 1]
```diff
@@ -3,8 +3,8 @@ const { WorkflowError } = require('../Errors');
 const fs = require('fs');
 const path = require('path');
 
-const generateMetadata = async (pinataClient, name, description, imageFile) => {
-  // pin image
+const generateMetadata = async (pinataClient, name, description, imageFile, videoFile) => {
+  // validate image
   let imagePath = path.resolve(imageFile);
```
Comment (hamidra) [inline src/workflow/metadata.js]: I think we should make image optional too.

File: src/workflow/workflow.js [hunk 2]
```diff
@@ -223,49 +223,72 @@ const pinAndSetImageCid = async (wfConfig) => {
...
+      videoCidColumn.records.push('');
+    }
     if (i >= metaCidColumn.records.length) {
       metaCidColumn.records.push('');
     }
 
     if (i >= startRecordNo && i < endRecordNo && !metaCidColumn.records[i]) {
-      let imageFileName = formatFileName(
-        fileNameTemplate,
+      const imageFileName = formatFileName(
```
Comment (hamidra) [inline src/workflow/workflow.js]: Image file should be optional too.

File: README.md [hunk 3]
```diff
@@ -69,6 +71,7 @@ Note:
 - The `instance.metadata.imageFile` specifies the file path that contains the media file that is going to be minted. 
 Path example: `/Users/user/nfts/<<image name>>.png` where `<<image name>>` will be replaced with the value in the "image name" column according to the row number in the csv datafile. 
 Additionally, you can use that template: `/Users/user/nfts/<>.png` in which the '<>' will be replaced with the row number.
```
Comment (hamidra) [inline README.md]: minor typo "this template".

**Review by hamidra — COMMENTED** [review]
I think image should be optional. So if someone want to just set video with no preview image.

**Comment by jsidorenko** [comment]
Done!

File: src/workflow/metadata.js [hunk 4]
```diff
@@ -3,47 +3,92 @@ const path = require('path');
...
+      throw new WorkflowError(`failed to pin video.`);
+    }
   }
 
   // create metadata
   let metadata = {
     name: name,
     image: `ipfs://ipfs/${imageCid}`,
-    description: description,
+    animation_url: videoCid ? `ipfs://ipfs/${videoCid}` : undefined,
```
Comment (hamidra) [inline src/workflow/metadata.js:70]: The image in metadata should be optional as well.

something like:
let metadata = {
    name,
    !!imageCid && ...{image: `ipfs://ipfs/${imageCid}`,
    !!videoCid && ...{animation_url: ? `ipfs://ipfs/${videoCid}`,
    description,
  };

Comment (jsidorenko) [reply src/workflow/metadata.js:70]: damn, I missed that. Good catch!

**Review by hamidra — APPROVED** [review]

**Pull request merged**